### PR TITLE
Adds 'flask' sub-command

### PR DESCRIPTION
**Pull request by tdstein**
## Intent

Adds the sub-command `flask` to `deploy` and `write-manifest.` 

In both instances, `flask` is syntactic sugar. The underlying implementation is a direct alias to the existing `api` sub-command.

## Type of Change

- [ ] Bug Fix            
- [x] New Feature       
- [ ] Breaking Change   

## Approach

The `api` implementation supports Flask APIs. Therefore, a new `AppMode` is not required.

This change adds an optional description parameter to the functions that construct the click argument. If provided, the value is used instead of `AppType#desc`.

## Directions for Reviewers

Try running `rsconnect deploy flask` and `rsconnect write-manifest flask`.

## Checklist

- [ ] I have updated [CHANGELOG.md](../CHANGELOG.md) to cover notable changes.
- [ ] I have updated all related GitHub issues to reflect their current state.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,10 +27,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   recognized filename patterns, the file patterns `app-*.py`, `app_*.py`, `*-app.py`,
   and `*_app.py` are now considered. However, if the directory contains more than
   one file matching these new patterns, you must provide rsconnect-python with an
-  explicit `--entrypoint` argument.
+  explicit `--entrypoint` argument.\
```
Comment (mmarchetti) [inline CHANGELOG.md:30]: nit: typo

File: README.md [hunk 2]
```diff
@@ -270,8 +270,9 @@ rsconnect write-manifest notebook my-notebook.ipynb
 You can deploy a variety of APIs and applications using sub-commands of the
 `rsconnect deploy` command.
 
-* `api`: WSGI-compliant APIs such as Flask and packages based on Flask
-* `fastapi`: ASGI-compliant APIs (FastAPI, Quart, Sanic, and Falcon)
+* `api`: WSGI-compliant APIs (e.g., `gunicorn`, `werkzeug`, `uwsgi`, `pyramid`, and `bottle`)
+* `flask`: Flask APIs
```
Comment (mmarchetti) [inline README.md:274]: `api` and `flask` should be synonyms - they're treated identically by Connect.

File: README.md [hunk 3]
```diff
@@ -270,8 +270,9 @@ rsconnect write-manifest notebook my-notebook.ipynb
 You can deploy a variety of APIs and applications using sub-commands of the
 `rsconnect deploy` command.
 
-* `api`: WSGI-compliant APIs such as Flask and packages based on Flask
-* `fastapi`: ASGI-compliant APIs (FastAPI, Quart, Sanic, and Falcon)
+* `api`: WSGI-compliant APIs (e.g., `gunicorn`, `werkzeug`, `uwsgi`, `pyramid`, and `bottle`)
```
Comment (mmarchetti) [inline README.md:273]: `gunicorn` and `uwsgi` I think are WSGI hosts, vs. frameworks for apps/APIs that can be deployed.

We have tested `bottle`, `flask-api`, `flask-restx`, and `flasgger`, but I don't think we have validated Pyramid.

There content we have integration tests for would be a good source of info here.

Comment (aronatkins) [reply README.md:273]: I don't know of pyramid tests, either.

On the runtime side, we definitely check:

* python-api: simple WSGI (no framework), bottle, falcon, pycnic
* python-dash: dash
* python-streamlit: streamlit
* python-bokeh: bokeh
* python-fastapi: fastapi, quart, sanic, starlette
* python-shiny: shiny

I may be missing some; @kgartland-rstudio has additional test coverage to the ones I'm looking at.

Comment (aronatkins) [reply README.md:273]: Oy. I just realized that my set of runtime tests doesn't include flask! 😭 I'll get that updated.

File: CHANGELOG.md (hunk 1 above)
Comment (tdstein) [reply CHANGELOG.md:30]: It took me over a minute to realize what the type is...

```suggestion
  explicit `--entrypoint` argument.
```

File: README.md (hunk 3 above)
Comment (tdstein) [reply README.md:273]: It seems that we should have explicit commands for each of the frameworks that we actively test. 

Until this week, I didn't know that `fastapi` supported other frameworks which implement the ASGI. I assume there are others like me.

This is outside of the scope of this issue, so we can have the conversation here, but I won't implement those changes here.

**Review by mmarchetti — APPROVED** [review]

**Pull request merged**